src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let amount = 0;
+
+  const adder = (arg) => {
+    if (arg !== undefined) {
+      amount += arg;
+
+      return adder;
+    }
+
+    const result = amount;
+
+    amount = 0;
+
+    return result;
+  };
+
+  return adder;
```

Comment:
The `makeInfinityAdder` function's inner `adder` function correctly accumulates the sum and resets it when called without arguments. However, there is a potential issue with the check for `undefined` on line 10. It's better to use a more robust check to ensure that the function behaves correctly when falsy values like `0` or `false` are passed as arguments. You could use the arguments object or check the arguments length to determine if the function was called without parameters.